### ISSUE TITLE
[bug] mlr stats1 prints nothing when two maybe-missing fields specified

### ISSUE THREAD

**Issue by Komosa** [johnkerl/miller#79]
input:
    
    a=1
    b=2

`cat input | mlr stats1 -a sum -f a`:
    
    a_sum=1.000000

but `cat input | mlr stats1 -a sum -f a,b` gives empty output

**Comment by johnkerl** [comment]
This is intentional -- it ignores records unless they have *all* aggregation fields. But the design should be changed to handle records which have *any* of the specified fields.

**Comment by johnkerl** [comment]
Fix in head & will go out with the 2.3.2 release.

```
$ cat input
a=1
b=2

$ cat input | mlr stats1 -a sum -f a
a_sum=1.000000

$ cat input | mlr stats1 -a sum -f b
b_sum=2.000000

$ cat input | mlr stats1 -a sum -f a,b
a_sum=1.000000,b_sum=2.000000
```

Similarly affected are `mlr stats2` and `mlr step`. There is some separate work yet to be done on `mlr join`.

Thanks for reporting this -- it had been in the back of my brain for some time, but I hadn't gotten round to doing anything about it. :)

**Comment by johnkerl** [comment]
Above I wrote "There is some separate work yet to be done on `mlr join`" -- https://github.com/johnkerl/miller/issues/82.
